### PR TITLE
Upgrade docker setup:

### PR DESCRIPTION
**Pull request by IceRidder**
## Description

*Update docker to php:8.1.0
*Install composer and dependencies in the docker image
*Remove "vendor/bin/phpunit" command from the image to allow installing composer dependencies from fresh
*Add composer install example command
*Update docker-compose commands to use "-u $(id -u ${USER}):$(id -g ${USER})" to have files generated in docker have the proper permission on the host machine
*Clean php80 commands*
*Add volume (commented by default) to take advantage of local composer cache

## Motivation and context

Why is this change required? What problem does it solve?

fixes#1

If it fixes an open issue, please link to the issue here (if you write `fixes #num`
or `closes #num`, the issue will be automatically closed when the pull is accepted.)

## How has this been tested?

Please describe in detail how you tested your changes.

Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our [continuous integration](http://www.phptherightway.com/#continuous-integration) server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [ x] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [ x] My pull request addresses exactly one patch/feature.
- [ x] I have created a branch for this patch/feature.
- [ x] Each individual commit in the pull request is meaningful.
- [ ] I have added tests to cover my changes. - this is not applicable as i did not touch code
- [x ] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,15 +1,16 @@
 # To use:
 # Run "docker-compose build" to rebuild the app container.
-# Run "docker-compose run --rm php80" to run the test script with xdebug.
-# Run "docker-compose run --rm php81" to run the test script on 8.1.
+# Run "docker-compose run -u $(id -u ${USER}):$(id -g ${USER}) --rm php81 composer install" to install dependencies.
+# Run "docker-compose run  -u $(id -u ${USER}):$(id -g ${USER})--rm php81 vendor/bin/phpunit" to run the test script on 8.1.
```
Comment (Crell) [inline docker-compose.yml:4]: I've read that there are ways to move that to a config file so it doesn't have to get re-entered each time you run the container.  Do you have a preferred one?  I'd much rather bake that in than make it a necessary extra part of an already long command.  (Though I suppose I could always through a Makefile at it.)

Comment (IceRidder) [reply docker-compose.yml:4]: I usually also use a Makefile, but i wanted to change as little as possible in this case. If you want i can push one also.
Also i did not update the composer.json, should i do that here or should i do a different ticket/PR?

Comment (Crell) [reply docker-compose.yml:4]: Makefile it is then.  Good to know I'm not the only one that still uses old reliable tools. :-)

**Comment by Crell** [comment]
Thanks!  I'll take care of the other tasks on this and related repos on Monday.

**Pull request merged**